web_ui/src/pages/create-project/components/pose-template/canvas/cursor-context-menu/cursor-context-menu.component.tsx

```diff
@@ -36,30 +30,45 @@ export const CursorContextMenu = ({
             event.preventDefault();
 
             if (isValidTrigger(event.target as Element)) {
+                const modal = document.querySelector('[data-testid="modal"]');
+
+                if (modal === null) return;
+
+                const modalBox = modal.getBoundingClientRect();
+
                 onOpen();
-                setCursorPosition({ x: event.clientX, y: event.clientY });
+                setCursorPosition({ x: event.clientX - modalBox.x + X_PADDING, y: event.clientY - modalBox.y });
             }
         },
         triggerRef
     );
 
-    return (
-        <OverlayContainer>
-            {isOpen && (
-                <ThemeProvider>
-                    <View
-                        ref={containerRef}
-                        position={'absolute'}
-                        top={cursorPosition.y}
-                        left={cursorPosition.x + X_PADDING}
-                        zIndex={100001}
-                        backgroundColor={'gray-200'}
-                        data-testid='position container'
-                    >
-                        {children}
-                    </View>
-                </ThemeProvider>
-            )}
-        </OverlayContainer>
+    const { overlayProps } = useOverlay(
+        {
+            isOpen: state.isOpen,
+            isDismissable: true,
+            shouldCloseOnBlur: false,
+            onClose: state.close,
+        },
+        ref
+    );
+
+    if (!state.isOpen) return null;
+
+    return createPortal(
+        <div ref={ref} {...overlayProps}>
+            <View
+                position={'absolute'}
+                top={cursorPosition.y}
+                left={cursorPosition.x}
+                zIndex={100001}
+                backgroundColor={'gray-200'}
+                data-testid='position container'
+                {...overlayProps}
+            >
+                {children}
+            </View>
+        </div>,
+        document.querySelector('[data-testid="modal"]') as HTMLElement
```

Comment:
It took me a moment to realize this selector refers to the parent modal, not the newly created portal elements. Let’s move this selector into a variable that can be used both at line 33 and here. `PARENT_MODAL_ELEMENT` could be an option